### ISSUE TITLE
[enhancement]sync changes from remarkable back to zotero

### ISSUE THREAD

**Issue by yongweiy** [samuelyeewl/zotrm#8]
I have been looking for a good way to manage papers on Remarkable tablet and I found zotrm quite handy. There is only one thing I wish the script could handle. Sometimes, we would make annotations on papers while reading on remarkable. Later on, it is very likely that we will turn to our laptop/PC to look for something (simply because it is easier to do). Therefore, I am wondering would that be great if we could bring changes on files from remarkable back to zotero. I am not quite familiar with the zotero api but I know python for sure and willing to take a look into the feature. Do you have any suggestion on where I should start with? @samuelyeewl

**Comment by samuelyeewl** [comment]
I agree, this is also something that I would like but haven't had time to work on. I imagine the following steps would need to be implemented:

- [ ] Reverse sync back to local computer.
    - [ ] Set a tag in Zotero to mark those PDFs that should be re-synced (or keep a running list somewhere).
    - [ ] Grab annotations from remarkable.
        - This might be done using `rsync` and a cron job (e.g. https://github.com/lucasrla/remarkable-utils), or
        - Using the rm API directly (not sure if it can grab the *.rm annotations).
- [ ] Add annotations to PDF
    - [ ] There are some libraries which can combined the annotations and base PDF into a new PDF file (e.g. [rMsync](https://github.com/lschwetlick/rMsync), [remarks](https://github.com/lucasrla/remarks), [rm-dl-annotated](https://github.com/jmptable/rm-dl-annotated)).
    - [ ] Personally, I would prefer if the annotations were added as actual PDF annotations, so they can be searched or deleted on the PC. There seem to be some projects focused on extracting the remarkable annotations (e.g. [biff](https://github.com/soulisalmed/biff), [remarkable-highlights](https://github.com/benlongo/remarkable-highlights)), which could be combined with e.g. [pdf-annotate](https://github.com/plangrid/pdf-annotate) to create native PDF annotations.
- [ ] Place annotated PDF back into Zotero, either as a new file or a separate one.

**Comment by yongweiy** [comment]
Thanks for the detailed information! But unfortunately, I am getting busy these days and may not have time soon. I will definitely come back to this later.

**Comment by samuelyeewl** [comment]
I've made a first pass at implementing this, using the [remarks](https://github.com/lucasrla/remarks) package.
Check out the `remarks` branch, you'll also need the `zotrm` branch in my fork of [remarks](https://github.com/samuelyeewl/remarks).

**Comment by tweedyf** [comment]
I just tried to use this to transfer a PDF from my Zotero library to my RM2, add a highlight, and sync back. I get an error on the return trip when zotrm.py tries to process the `stat` info from rmapi.

https://gist.github.com/tweedyflanigan/a0c6a77e7bc909f0b5224f33a902fb08
